src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/fs/TeeStepTest.java

```diff
@@ -76,6 +77,49 @@ public void smokes() throws Exception {
         });
     }
 
+    @Test
+    @Issue({"JENKINS-54346", "JENKINS-55505"})
+    public void closed() throws Exception {
+        rr.then(r -> {
+            r.createSlave("remote", null, null);
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(
+                    "node('remote') {\n" +
+                            "  tee('x.log') {\n" +
+                            "    echo 'first message'\n" +
+                            "  }\n" +
+                            "  if (isUnix()) {sh 'rm x.log'} else {bat 'del x.log'}\n" +
+                            "  writeFile file: 'x.log', text: 'second message'\n" +
+                            "  echo(/got: ${readFile('x.log').trim().replaceAll('\\\\s+', ' ')}/)\n" +
+                            "}", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            r.waitForCompletion(b);
+            r.assertLogContains("got: second message", b);
+        });
+    }
+
+    @Test
+    @Issue({"JENKINS-55505"})
+    public void closedMultiple() throws Exception {
+        rr.then(r -> {
+            r.createSlave("remote", null, null);
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(
+                    "node('remote') {\n" +
+                            "  tee('x.log') {\n" +
+                            "    if (!isUnix()) { bat 'echo first message' }\n" +
+                            "    if (!isUnix()) { bat 'echo second message' }\n" +
+                            "  }\n" +
+                            "  if (isUnix()) {sh 'rm x.log'} else {bat 'del x.log'}\n" +
+                            "  writeFile file: 'x.log', text: 'third message'\n" +
```

Comment:
Here again I am unclear on what exactly is being tested.

Reply:
This test the JENKINS-55505 (I think) ensuring that 2 writes using bat step doesn't lead to the tee output file being kept open. Ensures that it gets closed and can be deleted and overwritten.